### ISSUE TITLE
debian/Ubuntu postinst script should use invoke-rc.d to control service startup

### ISSUE THREAD

**Issue by kvick** [docker/docker#3924]
Debian postint scripts should use invoke-rc.d instead of service to control the start of services on package install based on the specifications in policy-rc.d.  See http://www.debian.org/doc/debian-policy/ch-opersys.html.

This is minor, and won't affect most users, but can cause issues when attempting to provide docker AMIs using tools such as https://github.com/netflix/aminator.

The symptom is it aminator fails when it tries to umount the volume as docker is running in the chroot immediately after the lxc-docker package install even with a policy-rc.d set to deny (the service call doesn't check policy-rc.d)

Something like

```
invoke-rc.d docker $_dh_action 2>/dev/null || true
```

instead of 

```
service docker $_dh_action 2>/dev/null || true
```

should work.  invoke-rc.d will still return a 0 status code if policy-rc.d is denied.

>
STATUS CODES
       Should an init script be executed, invoke-rc.d always returns the status code returned by the init script. Init scripts should not
       return status codes in the 100+ range (which is reserved in Debian and by the LSB).  The  status  codes  returned  by  invoke-rc.d
       proper are:
               0      Success.   Either the init script was run and returned exit status 0 (note that a fallback action may have been run instead
              of the one given in the command line), or it was not run because of runlevel/local policy constrains and --disclose-deny is
              not in effect.

**Comment by ncraike** [comment]
#dibs

**Comment by ncraike** [comment]
It looks like `hack/make/ubuntu`, around 103 is the code in question.

**Comment by ctrlrsf** [comment]
@ncraike I don't think this change addresses issue of calling "invoke-rc.d" instead of "service" on Debian. Debian has `service` command so `if` on line 104 would be true and therefore `_service_invoke` would be set to "service docker".

```
root@d5e940d6f554:/# cat /etc/debian_version 
8.0
root@d5e940d6f554:/# which service
/usr/sbin/service
```

**Comment by jfrazelle** [comment]
closing as this is really a stardards thing and since our script works for multiple distros now we cant really implement, but I am not belittling the standard we are working on changes to make this (per distro installs) happen and changed have been merged w #12111 aslso see https://github.com/docker/docker/pull/11808

**Comment by ncraike** [comment]
Agh, sorry @ctrlrsf. Had a busy few weeks and didn't get a chance to respond. In hindsight, I should've spun up Debian/Ubuntu containers and tested more.
